contributed_definitions/nyaml/NXfit_single_parameter.yaml

```diff
@@ -0,0 +1,83 @@
+category: base
+doc: |
+  A single parameter for a fit function.
+  This would typically be a variable that
+  is optimized in a fit.
+type: group
+NXfit_single_parameter(NXobject):
+  name(NX_CHAR):
+    doc: |
+      The name of the parameter.
+  description(NX_CHAR):
+    doc: |
+      A description of what this parameter represents.
+  value(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The value of the parameter.
+  min_value(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The minimal value of the parameter.
+  max_value(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The maximal value of the parameter.
```

Comment:
Does it make sense to add the option to define parameters as held fix during the fit?
And how can we express constraints between different parameters (maybe not necessary for now)

Reply:
I agree this is what we are missing here, not just constraints between parameters within one NXfit_function, but also e.g. "the area (a fit param) of peak1 should be half the area of peak2". Maybe we can add such constraints later.